satellitePropagatorExamples/SatellitePropagatorExamples/thrustAccelerationExample.cpp

```diff
@@ -27,171 +40,173 @@ int main()
+    bodyMap[ "Vehicle" ] = boost::make_shared< simulation_setup::Body >( );
 
     // Finalize body creation.
     setGlobalFrameBodyEphemerides( bodyMap, "SSB", "J2000" );
 
-    ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
-    ///////////////////////            CREATE ACCELERATIONS          //////////////////////////////////////////////////////
-    ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    /// CREATE ACCELERATIONS
 
     // Define propagator settings variables.
     SelectedAccelerationMap accelerationMap;
     std::vector< std::string > bodiesToPropagate;
     std::vector< std::string > centralBodies;
 
-    // Define propagation settings.
-    std::map< std::string, std::vector< boost::shared_ptr< AccelerationSettings > > > accelerationsOfAsterix;
-    accelerationsOfAsterix[ "Earth" ].push_back( boost::make_shared< SphericalHarmonicAccelerationSettings >( 5, 5 ) );
+    // Find filepath of this cpp file
+    std::string cppFilePath( __FILE__ );
 
-    accelerationsOfAsterix[ "Sun" ].push_back( boost::make_shared< AccelerationSettings >( 
-                                                   basic_astrodynamics::central_gravity ) );
-    accelerationsOfAsterix[ "Moon" ].push_back( boost::make_shared< AccelerationSettings >(
-                                                     basic_astrodynamics::central_gravity ) );
-    accelerationsOfAsterix[ "Mars" ].push_back( boost::make_shared< AccelerationSettings >(
-                                                     basic_astrodynamics::central_gravity ) );
-    accelerationsOfAsterix[ "Venus" ].push_back( boost::make_shared< AccelerationSettings >(
-                                                     basic_astrodynamics::central_gravity ) );
-    accelerationsOfAsterix[ "Sun" ].push_back( boost::make_shared< AccelerationSettings >(
-                                                     basic_astrodynamics::cannon_ball_radiation_pressure ) );
-    accelerationsOfAsterix[ "Earth" ].push_back( boost::make_shared< AccelerationSettings >(
-                                                     basic_astrodynamics::aerodynamic ) );
+    // Find folder of this cpp file
+    std::string cppFolder = cppFilePath.substr( 0 , cppFilePath.find_last_of("/\\")+1 );
+    Eigen::MatrixXd thrustForceMatrix =
+            input_output::readMatrixFromFile( cppFolder + "unitTests/nonconstantThrust.txt" , " \t", "#" );
+
+    // Make map for thrust data
+    std::map< double, Eigen::Matrix3d > thrustData;
+
+    // Fill thrustData map using thrustForceMatrix Eigen matrix
+    for ( int i = 0; i < thrustForceMatrix.rows(); i++ )
+    {
+        Eigen::Matrix3d blabla = thrustForceMatrix.rightCols( 3 ).row( i );
+        thrustData[ thrustForceMatrix( i, 0 ) ] = blabla;
+    }
+
+
+    // Define order of Lagrange interpolator
+    int interpolatorOrder = 3;
```

Comment:
The interpolation must be changed to an even number, e.g.:

`int interpolatorOrder = 2;`